platine-management-service/src/main/java/fr/insee/survey/datacollectionmanagement/questioning/service/impl/QuestioningAccreditationServiceImpl.java

```diff
@@ -39,9 +37,21 @@ public QuestioningAccreditation saveQuestioningAccreditation(QuestioningAccredit
         return questioningAccreditationRepository.save(questioningAccreditation);
     }
 
+    @Override
+    public QuestioningAccreditation findByQuestioningIdAndIsMain(Long questioningId) {
+        return questioningAccreditationRepository.findAccreditationsByQuestioningIdAndIsMainTrue(questioningId);
+    }
+
     @Override
     public void deleteAccreditation(QuestioningAccreditation acc) {
         questioningAccreditationRepository.deleteById(acc.getId());
     }
 
+    @Override
+    public void setMainQuestioningAccreditationToContact(String contactId, Long questioningId) {
+        QuestioningAccreditation questioningAccreditation = findByQuestioningIdAndIsMain(questioningId);
+        questioningAccreditation.setIdContact(contactId);
+        saveQuestioningAccreditation(questioningAccreditation);
```

Comment:
each time you add a QuestioningAccreditation, you need to add a row to the view table using viewService.createView(idContact, idSu, idCampaign)